src/vm/database/structures.rs

```diff
@@ -0,0 +1,84 @@
+use vm::contracts::Contract;
+use vm::errors::{Error, InterpreterError, RuntimeErrorType, UncheckedError, InterpreterResult as Result, IncomparableError};
+use vm::types::{Value, OptionalData, TypeSignature, TupleTypeSignature, AtomTypeIdentifier, PrincipalData, NONE};
+
+pub trait ClaritySerializable {
+    fn serialize(&self) -> String;
+}
+
+pub trait ClarityDeserializable<T> {
+    fn deserialize(json: &str) -> T;
+}
+
+macro_rules! clarity_serializable {
+    ($Name:ident) =>
+    {
+        impl ClaritySerializable for $Name {
+            fn serialize(&self) -> String {
+                serde_json::to_string(self)
```

Comment:
We should open an issue to guarantee that serialization of Clarity objects is stable, and bijective -- each Clarity object has exactly one byte-string representation.

Reply:
We can just add that as a requirement on this issue:

https://github.com/blockstack/blockstack-core/issues/946